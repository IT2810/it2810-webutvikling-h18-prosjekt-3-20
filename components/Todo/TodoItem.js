import React, { Component } from 'react';
import {
  View,
  Text,
  StyleSheet,
} from 'react-native';
import { CheckBox } from 'react-native-elements';
import PropTypes from 'prop-types';
import SwipeOut from 'react-native-swipeout';
import Colors from '../../constants/Colors';

const createStyles = completed => StyleSheet.create({
  container: {
    marginLeft: 10,
    marginRight: 10,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  checkbox: {
    flex: 1,
  },
  name: {
    flex: 5,
    color: completed ? Colors.completedTodo : Colors.black,
    textDecorationLine: completed ? 'line-through' : 'none',
  },
  date: {
    flex: 2,
    color: completed ? Colors.completedTodo : Colors.black,
    textDecorationLine: completed ? 'line-through' : 'none',
  },
});

const checkboxWrapperStyle = StyleSheet.create({
  container: {
    backgroundColor: Colors.transparent,
    borderWidth: 0,
    width: 40,
  },
});

export default class TodoItem extends Component {
  static propTypes = {
    onCheckBoxPress: PropTypes.func.isRequired,
    onRemoveTodo: PropTypes.func.isRequired,
    item: PropTypes.object.isRequired,
  };

  state = {
    // Keeps track of if/which todoItem is swiped left
    activeItem: null,
  };

  render() {
    const styles = createStyles(this.props.item.completed);
    // SwipeOut properties
    const swipeSettings = {
      autoClose: true,
      onOpen: () => {
        this.setState({ activeItem: this.props.item });
      },
      onClose: () => {
        if (this.props.item.id === this.state.activeItem && typeof direction !== 'undefined') {
          this.setState({ activeItem: null });
        }
      },
      right: [
        {
          // When pressing delete button
          onPress: () => {
            this.props.onRemoveTodo(this.state.activeItem);
          },
          text: 'Delete',
          type: 'delete',
        },
      ],
    };
    return (
      <SwipeOut {...swipeSettings}>
        <View style={styles.container}>
          <CheckBox
            style={styles.checkbox}
            checked={this.props.item.completed}
            containerStyle={checkboxWrapperStyle.container}
            onPress={() => {
              this.props.onCheckBoxPress(this.props.item);
            }}
          />
          <Text style={styles.name}>
            {this.props.item.name}
          </Text>
          <Text style={styles.date}>
            {this.props.item.date}
          </Text>
        </View>
      </SwipeOut>
    );
  }
}
