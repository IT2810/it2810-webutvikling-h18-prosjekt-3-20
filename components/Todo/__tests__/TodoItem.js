import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';
import TodoItem from '../TodoItem';

describe('<TodoItem/>', () => {
  it('should render empty component', () => {
    const tree = renderer.create(
      <TodoItem
        item={{}}
        onRemoveTodo={jest.fn()}
        onCheckBoxPress={jest.fn()}
      />,
    ).toJSON();
    expect(tree)
      .toMatchSnapshot();
  });
  it('should run onUpdate at CheckBox press', () => {
    const spy = jest.fn();
    const wrapper = shallow(<TodoItem onRemoveTodo={jest.fn()} onCheckBoxPress={spy} item={{}}/>);
    // console.log(wrapper.find('CheckBox'));
    wrapper.find('CheckBox').simulate('press');

    expect(spy).toHaveBeenCalled();
  });
});
