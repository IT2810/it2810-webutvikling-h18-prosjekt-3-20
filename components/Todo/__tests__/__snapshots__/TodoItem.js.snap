// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoItem/> should render empty component 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#dbddde",
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    onLayout={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {},
        Object {
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#ededed",
            "borderRadius": 3,
            "borderWidth": 0,
            "margin": 5,
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 1,
            "padding": 10,
            "width": 40,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              false,
              false,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            onLongPress={undefined}
            onPress={undefined}
            style={
              Array [
                Object {
                  "color": "#bfbfbf",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            ï‚–
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "#43484d",
                    "fontWeight": "bold",
                    "marginLeft": 10,
                    "marginRight": 10,
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
          />
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "flex": 5,
            "textDecorationLine": "none",
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "flex": 2,
            "textDecorationLine": "none",
          }
        }
      />
    </View>
  </View>
  <View />
  <View />
</View>
`;
